docs/design/drf - fairshare.md

```diff
@@ -0,0 +1,179 @@
+# Namespace fair share
+
+[@lminzhw](http://github.com/lminzhw); May 8, 2019
+
+## Motivation
+
+`Queue` was introduced in [kube-batch](http://github.com/kubernetes-sigs/kube-batch) to share resources among users.
+
+But, the user in the same `Queue` are equivalent during scheduling. For example, we have a `Queue` contains a small amount of resources, and there are 10 pods belong to UserA and 1000 pods belong to UserB. In this case, pods of UserA would have less probability to bind with node.
+
+So, we need a more fine-grained strategy to balance resource usage among users in the same `Queue`.
+
+In consideration of multi-user model in kubernetes, we use namespace to distinguish different user. Each namespace would have its weight to control resources usage.
+
+## Function Specification
+
+Weight have these features:
+> 1. `Queue` level
+> 2. an `integer` with default value 1
+> 3. record in namespace `quota`
+> 4. higher value means more resources after balancing
+
+### where is the weight
+
+```yaml
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  namespace: default
+spec:
+  hard:
+    limits.memory: 2Gi
+    volcano.sh/namespace.weight: 1  <- this field represent the weight of this namespace
+```
+
+If many `ResourceQuota` in the same namespace have weight, the weight for this namespace is the highest one of them.
```

Comment:
I don't quite understand the assumption here. Do you mean if user create several `ResourceQuota` in this namespace, the one with highest weight wins?

Reply:
Sure. Just for corner case.

Reply:
The hierarchy is `Queue` -> `PodGroup` -> `Pod` now. And there is no namespace now. It means that we could create pods in the same namespace for different queues. How do we deal with it?

Reply:
> It means that we could create pods in the same namespace for different queues

Yes, we support that; namespace is considered as a user, so we will balance the user's resources cross queue. If there's one queue per namespace, this feature will not take effect.